frontend/src/metabase/admin/settings/app/components/SettingsEditor/SettingsEditor.jsx

```diff
@@ -81,18 +81,28 @@ class SettingsEditor extends Component {
   }
 
   /**
-   * @param {Object} setting
+   * @param {Object|import("metabase-types/api").SettingKey} settingOrKey
    * @param {*} newValue
    * @param {Object} options - allows external callers in setting's that user custom components to hook into the success or failure of the update
    * @param {function} [options.onChanged] - callback fired after the setting has been updated
    * @param {function} [options.onError] - callback fired after the setting has failed to update
    */
-  updateSetting = async (setting, newValue, options) => {
-    const { settingValues, updateSetting, reloadSettings, dispatch } =
-      this.props;
+  updateSetting = async (settingOrKey, newValue, options) => {
+    const {
+      settingsByKey,
+      settingValues,
+      updateSetting,
+      reloadSettings,
+      dispatch,
+    } = this.props;
 
     this.saveStatusRef.current.setSaving();
 
+    const setting =
+      typeof settingOrKey === "string"
+        ? settingsByKey[settingOrKey]
+        : settingOrKey;
```

Comment:
this will be reverted as per Ryan's request to not complicate the work of settings. Before [Admin Settings Redesign](https://www.notion.so/metabase/Admin-Settings-Redesign-b8a9435f4c7d423399ac296d6b58cf1a) is done.